.github/workflows/CI.yml

```diff
@@ -58,12 +66,16 @@ jobs:
     runs-on: ${{ matrix.os }}
     steps:
       - name: "Checkout repository 🛎"
-        uses: actions/checkout@v4
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683
+        with:
+          persist-credentials: false
+
       - name: "Setup CI environment 🛠"
-        uses: ./.github/actions/set-dev-env
+        uses: pydata/pydata-sphinx-theme/.github/actions/set-dev-env@01731d0cc57768b9eff1c97f38909932ecd7e7d1
```

Comment:
here since it's our own action, I would have tended toward always using the latest version of it (I'm imagining my future self tweaking that action, and either not noticing or not knowing why my changes weren't manifesting in the CI jobs).  It also means that even if one did remember that this line needs to be updated, you couldn't do it until after you knew the commit hash on `main` of the PR that altered the action (which seems cumbersome: requires 2 PRs).


That said, if you have a strong reason for pinning to a commit, maybe add a comment in the action that this pin exists, so that future maintainers have a reminder to bump the pin after they make changes.

Reply:
Ok in that case let's just add a comment to the action itself reminding folks to bump the pin after any changes to it are merged.